pkg/app/types.go

```diff
@@ -61,6 +56,21 @@ type InstanceManager interface {
 	// original method returns no data on success, such as `Delete`, response will be empty. If the original method
 	// is standard `Get`/`Create`/`Update`, the response should be the relevant resource.
 	WaitOperation(zone string, user UserInfo, name string) (any, error)
+	// Creates a connector to the given host.
+	GetHostClient(zone string, host string) (HostClient, error)
+}
+
+type HostClient interface {
+	// Get and Post requests return the HTTP status code or an error.
+	// The response body is parsed into the res output parameter if provided.
+	Get(URLPath, URLQuery string, res *HostResponse) (int, error)
+	Post(URLPath, URLQuery string, bodyJSON any, res *HostResponse) (int, error)
+	GetReverseProxy() *httputil.ReverseProxy
+}
+
+type HostResponse struct {
+	responseObj   any
```

Comment:
nit, let's make these fileds public, also remove `response` prefix which is redundant.

Reply:
Done